src/worker.rs

```diff
@@ -46,18 +43,23 @@ impl<'a> SidekiqWorker<'a> {
                tx: Sender<Signal>,
                rx: Receiver<Operation>,
                queues: Vec<String>,
-               weights: Vec<f64>,
-               handlers: BTreeMap<String, Box<JobHandler>>,
-               middlewares: Vec<Box<MiddleWare>>,
+               handlers: BTreeMap<String, Box<dyn JobHandler>>,
+               middlewares: Vec<Box<dyn MiddleWare>>,
                namespace: String)
                -> SidekiqWorker<'a> {
+
+        let mut rng = rand::thread_rng();
+        let identity: Vec<u8> = iter::repeat(()).map(|()| rng.sample(distributions::Alphanumeric)).take(9).collect();
+
+        let mut queues = queues.clone();
+        queues.shuffle(&mut rand::thread_rng());
```

Comment:
The previous code passed the queue weights down to each worker, and relied on the fairly complex [`random_choice_f64`](https://github.com/StefanoD/Rust_Random_Choice/blob/ca6b8bf3b97a82e4c7bdeac236b609b196840972/src/lib.rs#L87) to combine the two `Vec`s. That crate is unmaintained I opted to mirror the Ruby implementation.

First it adds duplicate entries based on the queue weight:

https://github.com/mperham/sidekiq/blob/1e7ace608b5e10f77133ee21c2d70671a413fee9/lib/sidekiq/cli.rb#L418

Then it shuffles the array for every worker:

https://github.com/mperham/sidekiq/blob/87b2c4742300a3fc1b2c69a04e30e5e769df6783/lib/sidekiq/fetch.rb#L84

There is one difference to note: it looks like the Ruby version shuffles the array every time when polling for new jobs, but I can't think of a reason why that would make a difference, so instead the Rust version shuffles queues once at worker boot.